projects/storefrontlib/src/cms-components/cart/add-to-cart/add-to-cart.component.spec.ts

```diff
@@ -122,6 +128,24 @@ describe('AddToCartComponent', () => {
       expect(addToCartComponent.hasStock).toEqual(true);
     });
 
+    it('should reset counter value when changing product', () => {
+      //Product 1
+      let spy = spyOn(currentProductService, 'getProduct').and.returnValue(
+        of(mockProduct)
+      );
+      addToCartComponent.ngOnInit();
+      expect(addToCartComponent.productCode).toEqual(mockProduct.code);
+      addToCartComponent.quantity = 5;
+
+      //Product 2
+      addToCartComponent.productCode = undefined;
+      spy.and.returnValue(of(mockProduct2));
+      addToCartComponent.ngOnInit();
```

Comment:
Just FYI how to mock a service's stream with BehaviorSubject and avoid calling `ngOnInit` multiple times (which creates a new subscription each time):
```typescript
const currentProduct = new BehaviorSubject(mockProduct); // create a mock stream and pass initial value to emit
spyOn(currentProductService, 'getProduct').and.returnValue(currentProduct);
addToCartComponent.ngOnInit(); // call once
// ...
currentProduct.next(mockProduct2); // emit new value from the mock stream; the subscription from ngOnInit fires again itself
// ...
```